### PR TITLE
[themes] Fixup: use valid XML attribute format for 'defer' when loading searchindex file.

### PR DESCRIPTION
**Pull request by jayaddison**
### Feature or Bugfix
- Experiment / for consideration.

### CODE REVIEW

File: sphinx/themes/basic/search.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
     <script src="{{ pathto('_static/language_data.js', 1) }}"></script>
 {%- endblock %}
 {% block extrahead %}
-    <script src="{{ pathto('searchindex.js', 1) }}" defer></script>
+    <script src="{{ pathto('searchindex.js', 1) }}" defer="true"></script>
```
Comment (jayaddison) [inline sphinx/themes/basic/search.html:18]: @picnixz our `search.html` file is valid XHTML apart from this line, so this fixes parsing using `xml.etree`.

Comment (picnixz) [reply sphinx/themes/basic/search.html:18]: Technically... yes but without an attribute value it should be fine as well for HTML 5. So I'm wondering whether we want to use HTML5 specs or make it XHTML.

Comment (jayaddison) [reply sphinx/themes/basic/search.html:18]: Ok, good point.  I didn't realize that HTML5 specifies that [boolean attributes should not contain values](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute).

> [!NOTE]
> The values "true" and "false" are not allowed on boolean attributes. To represent a false value, the attribute has to be omitted altogether.

Comment (jayaddison) [reply sphinx/themes/basic/search.html:18]: ...continuing to the next sentence, though:

> If the attribute is present, its value must either be the empty string or a value that is an ASCII case-insensitive match for the attribute's canonical name, with no leading or trailing whitespace.

So there _is_ a way to achieve both XML and HTML5 compliance here.

**Pull request closed without merging**